src/main/java/org/thoughtcrime/securesms/ApplicationContext.java

```diff
@@ -131,15 +131,16 @@ public void onCreate() {
 
     rpc.start();
 
-    // migrating chat backgrounds, added  04/10/23, can be removed after some versions
-    String backgroundImagePath = Prefs.getStringPreference(this, "pref_chat_background", "");
-    if (!backgroundImagePath.isEmpty()) {
+    // migrating global notifications pref. to per-account config, added  10/July/24
+    final String NOTIFICATION_PREF = "pref_key_enable_notifications";
+    boolean isMuted = !Prefs.getBooleanPreference(this, NOTIFICATION_PREF, true);
+    if (isMuted) {
       for (int accId : dcAccounts.getAll()) {
-        Prefs.setBackgroundImagePath(this, accId, backgroundImagePath);
+        dcAccounts.getAccount(accId).setMuted(true);
       }
-      Prefs.setStringPreference(this, "pref_chat_background", "");
+      Prefs.removePreference(this, NOTIFICATION_PREF);
```

Comment:
as `SwitchPreferenceCompat` (resp. its base classes)  persist changes in "shared preferences" under the name `pref_key_enable_notifications` as well, the migration is executed whenever the user muted notifications for any account.

on the next app start, all accounts get muted then.

i do not know if we can prevent `SwitchPreferenceCompat` from persisting things, an easy fix would be to use a different name as `pref_enable_notifications` (strike the "key")

Reply:
good catch! I renamed it as per your suggestion

Reply:
the references need to be changed as well (i'd do a grep over .xml and .java and rename everything but the migration)

Reply:
I fixed it, I wonder if it is possible to move such keys to some xml file and then don't have this problem, it is easy to overlook and the keys are manually typed in both sides

Reply:
for me, there are still dependencies in `preferences_notifications.xml` as `android:dependency="pref_key_enable_notifications"` , this leads to crashes on opening the page.

but maybe a commit is missing again ...

Reply:
sorry, fixed it and compiled locally this time before pushing it (was thinking it was a quick simple single-line renaming fix and didn't test last times)